src/index.js

```diff
@@ -142,32 +142,129 @@ export default class Undo {
   }
 
   /**
-   * Decreases the current position and renders the data in the editor.
+   * insert a block deleted previously
+   * @param {Array} state is the current state according to this.position.
+   * @param {Array} compState is the state to compare and know the deleted block.
+   * @param {Array} index is the block index in state.
+   */
+  insertDeletedBlock(state, compState, index) {
+    for (let i = 0; i < state.length; i += 1) {
+      if (!compState[i] || state[i].id !== compState[i].id) {
+        this.editor.blocks.insert(state[i].type, state[i].data, {}, i, true);
+        this.editor.caret.setToBlock(index, 'end');
+        break;
+      }
+    }
+  }
+
+  /**
+   * return true if a block was dropped previously
+   * @param {Array} state is the current state according to this.position.
+   * @param {Array} compState is the state to compare and know the dropped block.
+   * @returns true if the block was dropped
+   */
+  blockWasDropped(state, compState) {
+    if (state.length === compState.length) {
+      return state.some((block, i) => block.id !== compState[i].id);
+    }
+    return false;
+  }
+  /**
+   * return true if the block has to be deleted becuase it was skipped previously.
+   * @param {*} index is the block index in state.
+   * @param {*} compIndex is the index to compare and know if the block was inserted previously.
+   * @returns true if a block was inserted previously and it has to be deleted.
+   */
+
```

Comment:
@MarioRodriguezS Could you please format this code fragment and remove this space